lib/system/web/src/keyboard_engine.rs

```diff
@@ -0,0 +1,170 @@
+            bindings,
+            _element: element.clone(),
+            _prevent_default: prevent_default,
+            _state: state,
+
+            //keydown_cb: keydown_cb,
+            //keyup_cb: keyup_cb,
+        }
+    }
+
+    pub fn from_tag_name(html_tag: &str, prevent_default: bool) -> Self {
+        let window = web_sys::window().expect("no global window exists");
+        let document = window.document().expect("should have document on window");
+        let element = document.get_elements_by_tag_name(html_tag).item(0).expect("there is no such element");
+
+        KeyboardEngine::new(&element, prevent_default)
+    }
+
+    fn init (
+        element:         &web_sys::Element,
+        bindings:        Rc<RefCell<Bindings>>,
+        state:           Rc<RefCell<BTreeSet<String>>>,
+        prevent_default: bool
+    //) -> (
+    //    Closure<dyn FnMut(web_sys::KeyboardEvent)>,
+    //    Closure<dyn FnMut(web_sys::KeyboardEvent)>
+    ) {
+        let capture_key_cb = Rc::new(Box::new(move |mut code: String, action: Action| {
+            code.make_ascii_lowercase();
+            match action {
+                Action::KeyDown => {
+                    state.borrow_mut().insert(code);
+                    let key = state
+                        .borrow()
+                        .iter()
+                        .join("-");
+
+                    if let Some(f) = bindings.borrow_mut().get_mut(&key) { f(); }
+                },
+                Action::KeyUp => {
+                    state.borrow_mut().remove(&code);
+                }
+            };
+
+        }));
+
+        let keydown_cb = {
+            let capture_key_cb = capture_key_cb.clone();
+            Closure::wrap(Box::new(move |event: web_sys::KeyboardEvent| {
+                capture_key_cb(event.key(), Action::KeyDown);
+                if prevent_default {
+                    event.prevent_default();
+                }
+            }) as Box<dyn FnMut(_)>)
+        };
+
+        let keyup_cb = Closure::wrap(Box::new(move |event: web_sys::KeyboardEvent| {
+            capture_key_cb(event.key(), Action::KeyUp);
+            if prevent_default {
+                event.prevent_default();
+            }
+        }) as Box<dyn FnMut(_)>);
+
+        element.add_event_listener_with_callback("keydown", keydown_cb.as_ref().unchecked_ref()).unwrap();
+        element.add_event_listener_with_callback("keyup", keyup_cb.as_ref().unchecked_ref()).unwrap();
+        // TODO: without forget the events doesnt work in Browser
+        keydown_cb.forget();
+        keyup_cb.forget();
+
+        //return (keydown_cb, keyup_cb);
+    }
+
+
+    /// Capture key press combination
+    /// combination has notation as <C-m> <M-Space> where
+    ///   C/Control - Contorl
+    ///   M/Meta - Meta
+    ///   A/Alt - Alt
+    ///   S/Shift - Shift
+    ///   Space
+    ///   a-z - alphabetic letters
+    pub fn capture(&mut self, combination: String, callback: Box<dyn FnMut()>) {
```

Comment:
Using `combination` as String is not the best idea here as well. We need some custom parser, which will always be slow and error-prone. Why not `keyboard_engine.capture([key::Control, key::A], ...)` instead? :) 

It could be easily changed to bitfield mask inside of the function!

Reply:
I've reworked the section, but would like to note that `keyCode` is [deprecated](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode)
[KeyboardEventInit section](https://www.w3.org/TR/uievents/#idl-keyboardeventinit)

> In practice, keyCode and charCode are inconsistent across platforms and even the same implementation on different operating systems or using different localizations. This specification does not define values for either keyCode or charCode, or behavior for charCode. In conforming UI Events implementations, content authors can instead use key and code.

but `key` and `code` are strings